searx/static/themes/__common__/js/image_layout.js

```diff
@@ -136,6 +147,11 @@
     var results_nodes = d.querySelectorAll(this.results_selector);
     var results_length = results_nodes.length;
 
+    function img_load_error(event) {
+      // console.log("ERROR can't load: " + event.originalTarget.src);
+      event.originalTarget.src = window.searxng.static_path + 'img/img_load_error.svg';
```

Comment:
Nit: `window.searxng.static_path` --> `w.searxng.static_path` (`w = window`, see at the end of the file)

---

In the Oscar theme, the image won't be found.

A solution: create a not generate d file in `img/img_load_error.svg` in the Oscar theme.

Another solution is to make the image optional with an additional parameter `errorImgUrl`:

```js
function ImageLayout(container_selector, results_selector, img_selector, verticalMargin, horizontalMargin, maxHeight, errorImgUrl) {
    this.errorImgUrl = errorImgUrl

    ...
    function img_load_error(event) {
      // console.log("ERROR can't load: " + event.originalTarget.src);
      event.originalTarget.src = this.errorImgUrl
    }

    ...

    if (this.errorImgUrl) {
        img.addEventListener('error', img_load_error, {once: true});
    }
}
```

then replace
https://github.com/searxng/searxng/blob/0311eba538df589bc11a4d76d3c94b104c5b764d/searx/static/themes/simple/src/js/main/results.js#L6
by
```js
searxng.image_thumbnail_layout = new searxng.ImageLayout('#urls', '#urls .result-images', 'img.image_thumbnail', 14, 6, 200, searxng.static_path + 'img/img_load_error.svg');
```

Reply:
> Another solution is to make the image optional with an additional parameter errorImgUrl:

I picked up this your suggestion, but instead of adding (more) arguments to the ImageLayout constructor, I added the `window.searxng.theme` namespace which can be passed through to the JS implementations / 7dc9cc9

IMO we should have one place (_single point of definition_) where we set all the theme values (parameters) for the JS client side.  If you think it is a good idea to have `window.searxng.theme` we can move `verticalMargin`, `horizontalMargin`, `maxHeight` and may be other paramaters as well to this namespace.

Reply:
* When #368 will be implemented, `searx/static/themes/__common__/js/image_layout.js` is going to be removed. The margins, widths will be CSS variables. There is one exception to this: the error handling. It can be implemented as a function in `searx/static/themes/simple/src/js/main/results.js`.
* `searx/static/themes/simple/src/js/head/00_init.js` exists only for one reason : it changes some CSS classes. If this change is made at the end of the HTML file, for a very short time, the HTML page is displayed without these changes, which is uncomfortable.
* --> Why not for a central place, but I wish to avoid the `searx/static/themes/simple/src/js/head/00_init.js` location: the browser loads this file in the `<head>` section = the larger the file, the slower the response time. One variable won't change the response time, but if we move all the theme constant to this place, I don't know how long the list will be. 

Dumb idea: why not CSS variables and then read them from Javascript? We could change the image according to the theme, media queries etc for free. See https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties#values_in_javascript ?